### PR TITLE
Add regex to check ip/url

### PR DESCRIPTION
**Pull request by Tony763**
#### Description
PR for issue #79 
Adds regex to check ip/url supplied to settings.

#### Type of PR
- [ ] Bugfix
- [x] Feature implementation
- [ ] Refactor of code (without functional changes)
- [ ] Documentation improvements
- [ ] Test improvements

#### Testing
[https://regex101.com/r/rGpkEo/1](https://regex101.com/r/rGpkEo/1)

#### Thanks
Regex based on [https://stackoverflow.com/a/50790119/9442005](https://stackoverflow.com/a/50790119/9442005) by Allan

### CODE REVIEW

**Comment by Tony763** [comment]
Hi @krisgesling, check please :slightly_smiling_face:

File: ha_client.py [hunk 1]
```diff
@@ -9,6 +10,28 @@
...
+                   r',1}[0-9])\.){3,3}(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]){0,1}[0-9])|'
+                   r'(?:[0-9a-fA-F]{1,4}:){1,4}:(?:(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]'
+                   r'){0,1}[0-9])\.){3,3}(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]){0,1}[0-9'
+                   r']))))(?::[0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{'
+                   r'2}|655[0-2][0-9]|6553[0-5])?(?:/[\w\.-]*)*/?\b'))
+
+def check_url(ip):
+    """Fuction to check if valid url/ip was supplied"""
+    matches = re.search(ip_regex,ip)
+    return matches
```
Comment (krisgesling) [inline ha_client.py]: Should we only be returning `matches.group(1)`? 

If returning the whole match object, can we describe what the groups are in the functions docstring?

File: ha_client.py [hunk 2]
```diff
@@ -9,6 +10,28 @@
...
+                   r'4}:(?:(?::[0-9a-fA-F]{1,4}){1,6})|:(?:(?::[0-9a-fA-F]{1,4}){1,'
+                   r'7}|:)|fe80:(?::[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(?:fff'
+                   r'f(?::0{1,4}){0,1}:){0,1}(?:(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]){0'
+                   r',1}[0-9])\.){3,3}(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]){0,1}[0-9])|'
+                   r'(?:[0-9a-fA-F]{1,4}:){1,4}:(?:(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]'
+                   r'){0,1}[0-9])\.){3,3}(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]){0,1}[0-9'
+                   r']))))(?::[0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{'
+                   r'2}|655[0-2][0-9]|6553[0-5])?(?:/[\w\.-]*)*/?\b'))
+
+def check_url(ip):
```
Comment (krisgesling) [inline ha_client.py:31]: Can we add a few test cases to a unittest for this function? 

Just thinking about anyone looking at it in the future and I wouldn't be confident that the regex101 link would remain active indefinitely.

**Review by krisgesling — COMMENTED** [review]
Damn - that is one hell of a regex!  Looks like it does exactly what we want though.

File: ha_client.py (hunk 1 above)
Comment (Tony763) [reply ha_client.py]: Group 1 will be sufficient. Will add.

File: ha_client.py (hunk 2 above)
Comment (Tony763) [reply ha_client.py:31]: True. I am making a unittests working again in my GA PR. When done, I will add it to this too.

File: ha_client.py (hunk 1 above)
Comment (Tony763) [reply ha_client.py]: Added.

File: ha_client.py (hunk 2 above)
Comment (Tony763) [reply ha_client.py:31]: Added for IP4, IP6, and domain/host name.

**Comment by Tony763** [comment]
@krisgesling Should be ready :slightly_smiling_face:

**Review by krisgesling — APPROVED** [review]
Looks great, thanks!

**Pull request merged**